pipeline/transformer/transformer_common/src/main/scala/weco/pipeline/transformer/identifiers/SourceIdentifierValidation.scala

```diff
@@ -0,0 +1,65 @@
+package weco.pipeline.transformer.identifiers
+
+import grizzled.slf4j.Logging
+import weco.catalogue.internal_model.identifiers.{
+  IdentifierType,
+  SourceIdentifier
+}
+
+import java.util.UUID
+import scala.Function.const
+import scala.util.Try
+import scala.util.matching.Regex
+
+object SourceIdentifierValidation {
+  implicit class SourceIdentifierOps(sourceIdentifier: SourceIdentifier)
+      extends Logging {
+    import IdentifierRegexes._
+
+    def validated: Option[SourceIdentifier] =
+      if (isValid) {
+        Some(sourceIdentifier)
+      } else {
+        None
+      }
+
+    def validatedWithWarning: Option[SourceIdentifier] = validated match {
+      case None =>
+        warn(
+          s"Invalid ${sourceIdentifier.identifierType.id}: ${sourceIdentifier.value}"
+        )
+        None
+      case some => some
+    }
```

Comment:
Is there ever a reason I wouldn't want a warning of an invalid identifier? Why does this exist separate from `.validated`?

Reply:
Yes - sometimes it's fine/expected for a particular type to be invalid, as we can fall back (ie for Miro library references)